android/src/main/java/com/balthazargronon/RCTZeroconf/ZeroconfModule.java

```diff
@@ -115,32 +79,91 @@ protected void sendEvent(ReactContext reactContext,
+        String pattern = "^[^.]*";
+        Pattern r = Pattern.compile(pattern);
+
+        Matcher m = r.matcher(fqdn);
+
+        if (m.find()) {
+            return m.group(0);
+        }
+
+        return fqdn;
+    }
+
+    private class MyDiscoverResolver extends DiscoverResolver {
+        MyDiscoverResolver(Context context, String serviceType, int debounceMillis) {
+            super(context, serviceType, debounceMillis);
+        }
+
         @Override
-        public void onResolveFailed(NsdServiceInfo serviceInfo, int errorCode) {
-            if (errorCode == NsdManager.FAILURE_ALREADY_ACTIVE) {
-                mNsdManager.resolveService(serviceInfo, this);
-            } else {
-                String error = "Resolving service failed with code: " + errorCode;
-                sendEvent(getReactApplicationContext(), EVENT_ERROR, error);
-            }
+        public void onDiscoveryStarted() {
+            sendEvent(getReactApplicationContext(), EVENT_START, null);
         }
 
         @Override
-        public void onServiceResolved(NsdServiceInfo serviceInfo) {
+        public void onDiscoveryStopped() {
+            sendEvent(getReactApplicationContext(), EVENT_STOP, null);
+        }
+
+        @Override
+        public void onServiceFound(NsdServiceInfo serviceInfo) {
             WritableMap service = new WritableNativeMap();
             service.putString(KEY_SERVICE_NAME, serviceInfo.getServiceName());
-            service.putString(KEY_SERVICE_FULL_NAME, serviceInfo.getHost().getHostName() + serviceInfo.getServiceType());
-            service.putString(KEY_SERVICE_HOST, serviceInfo.getHost().getHostName());
-            service.putInt(KEY_SERVICE_PORT, serviceInfo.getPort());
+
+            sendEvent(getReactApplicationContext(), EVENT_FOUND, service);
+        }
+
+        @Override
+        public void onServiceLost(MDNSDiscover.Result serviceResult) {
+            WritableMap service = new WritableNativeMap();
+            service.putString(KEY_SERVICE_NAME, getServiceName(serviceResult.srv.fqdn));
+            service.putString(KEY_SERVICE_FULL_NAME, serviceResult.srv.fqdn);
+            sendEvent(getReactApplicationContext(), EVENT_REMOVE, service);
+        }
+
+        @Override
+        public void onServiceResolved(MDNSDiscover.Result serviceResult) {
+            WritableMap service = new WritableNativeMap();
+            service.putString(KEY_SERVICE_NAME, getServiceName(serviceResult.srv.fqdn));
+            service.putString(KEY_SERVICE_FULL_NAME, serviceResult.srv.fqdn);
+            service.putString(KEY_SERVICE_HOST, serviceResult.srv.target);
+            service.putInt(KEY_SERVICE_PORT, serviceResult.srv.port);
+
+            WritableMap txt = new WritableNativeMap();
+            for (Map.Entry<String, String> entry : serviceResult.txt.dict.entrySet()) {
+               txt.putString(entry.getKey(), entry.getValue());
+            }
+            
+            service.putMap(KEY_SERVICE_TXT, txt);
```

Comment:
Can you give me an example output? Could be good to add in the readme, once we have a matching iOS api.